### PR TITLE
remove dependency on react-foundation-components library, refs #2253

### LINKED ISSUE #2253: Remove  react-foundation-components and write voting mechanism test
Currently blocking upgrade to React-16 and poorly maintained lib.

https://github.com/aruberto/react-foundation-components

### LINKED ISSUE #2303: Top level dropdowns do not accept theme-ing.
Condenser's current top-level dropdowns exist outside the theming context. They cannot respond to 'night mode' theme changes (or any others for that matter)

Though theme-dark is set, the dropdown exists outside the regular tree.
screen shot 2018-01-08 at 2 51 19 pm

Two possible solutions (of many) add the theme class to either the `<body>` tag, or the root of the tree that holds the dropdown.

body:
screen shot 2018-01-08 at 2 53 06 pm
dropdown root:
screen shot 2018-01-08 at 2 50 52 pm

### PR DESCRIPTION
**Pull request by gl2748**
Fixes #2253 
Fixes #2303 
The `react-foundation-components` library is a bug-bear, inconsistently maintained and unpredictable.
We'd rather not use it, but unfortunately it is used extensively our codebase.

This PR zaps the `react-foundation-components` dependency aiming for parity where possible, and subject to review, with a couple beneficial changes - namely:

### Before:
before_real

### After:
after

### Before.
b4_flag

### After.
after_flag2

ToDo.
- [x] Look through the wording for the flag action / modal text.
- [x] PR review feedback.
- [x]  Product feedback.

### CODE REVIEW

**Comment by gl2748** [comment]
# 1. First Flag Action - Flagging content with no prior interaction.
first_flag_action

# 2. Second  Flag Action - Unflagging previously flagged content.
second_flag_action

# 3. Third Flag Action - Voting up previously flagged content.
third_flag_action

#4. Forth Flag Action - Flagging previouslt voted-up content.
forth_flag_action

**Comment by gl2748** [comment]
@originated would it be possible to get a translation for this text?
https://github.com/steemit/condenser/pull/2434/files#diff-546d10b688dcc6d28fdf6e25bb85f49dR39

**Comment by bnchdrff** [comment]
@gl2748 just add it to en.json & wrap it in tt()

**Comment by bnchdrff** [comment]
wrap the token you create *

**Comment by bnchdrff** [comment]
@gl2748 ping

**Comment by gl2748** [comment]
# Voting Slider Demos
# UP
vote_up_slider

# FLAG
flag_slider

File: package.json [hunk 1]
```diff
@@ -120,6 +119,7 @@
     "redux": "3.7.2",
     "redux-form": "5.3.4",
     "redux-saga": "0.9.5",
+    "regexp-quote": "^0.0.0",
```
Comment (bnchdrff) [inline package.json]: porque?

Comment (gl2748) [reply package.json]: Idk, this morning this was happening... 
https://github.com/punkave/sanitize-html/issues/205

File: package.json [hunk 2]
```diff
@@ -100,15 +100,14 @@
...
-    "react-dropzone": "3.13.4",
-    "react-foundation-components": "git+https://github.com/valzav/react-foundation-components.git#d14362c7c8eee946a4acc3b18d70271d5a82813e",
-    "react-highcharts": "8.4.2",
-    "react-intl": "2.4.0",
-    "react-medium-editor": "1.8.1",
-    "react-notification": "5.0.7",
-    "react-overlays": "0.7.2",
-    "react-portal": "2.2.1",
-    "react-prop-types": "0.3.2",
+    "react-dropzone": "^3.7.3",
```
Comment (bnchdrff) [inline package.json]: please keep the corrected & fixed dependency versions

Comment (gl2748) [reply package.json]: oops yes.

**Comment by gl2748** [comment]
# IE Screen caps
screenshot from 2018-02-08 11-03-46
screenshot from 2018-02-08 11-04-08
screenshot from 2018-02-08 11-04-28

File: package.json [hunk 3]
```diff
@@ -101,7 +101,6 @@
     "react-copy-to-clipboard": "4.3.1",
     "react-dom": "15.6.2",
     "react-dropzone": "3.13.4",
-    "react-foundation-components": "git+https://github.com/valzav/react-foundation-components.git#d14362c7c8eee946a4acc3b18d70271d5a82813e",
```
Comment (bnchdrff) [inline package.json]: we also need to remove `foundation-sites` as well as the reference to it in `src/app/assets/stylesheets/app.scss`

Comment (gl2748) [reply package.json]: I think we still use foundation-sites css. This is just removing the 'react-foundation-components' library.

I think I might be remiss in suggesting we should remove foundation-sites.

Comment (gl2748) [reply package.json]: A seperate issue might be to just mode the css we're using in there over into condenseer, but I think that's a seperate issue...

File: src/app/components/elements/Voting.jsx [hunk 4]
```diff
@@ -141,7 +152,6 @@ class Voting extends React.Component {
             }
             this.setState({ showWeight: !this.state.showWeight });
         };
-        this.shouldComponentUpdate = shouldComponentUpdate(this, 'Voting');
```
Comment (bnchdrff) [inline src/app/components/elements/Voting.jsx]: is this component is `connect`ed with react redux & that is doing an acceptable job of managing `shouldComponentUpdate`?

File: src/app/components/pages/UserProfile.jsx [hunk 5]
```diff
@@ -552,21 +552,13 @@ export default class UserProfile extends React.Component {
...
-                                    {tt('g.rewards')}
-                                    <Icon name="dropdown-arrow" />
-                                </a>
-                            </LinkWithDropdown>
-                        </li>
+                        <DropdownMenu
+                            className={rewardsClass}
+                            items={rewardsMenu}
+                            el="li"
+                            selected={tt('g.rewards')}
```
[the remark this answers is not in the document]
Comment (gl2748) [reply src/app/components/pages/UserProfile.jsx]: Dropdown menu will look for labels and values to determine what is selected:

https://github.com/steemit/condenser/blob/3ed0ed5633097fbe732343bbc766c32b6167b2ca/src/app/components/elements/DropdownMenu.jsx#L63-L70

Furthermore - in this case, both the label and value of the items in this menu are translated:

https://github.com/steemit/condenser/blob/3ed0ed5633097fbe732343bbc766c32b6167b2ca/src/app/components/pages/UserProfile.jsx#L507-L518

File: src/app/components/cards/PostFull.scss [hunk 6]
```diff
@@ -121,12 +121,6 @@
         fill: themed('textColorSecondary');
     }
   }
-  a, .FoundationDropdownMenu__label {
```
[the remark this answers is not in the document]
Comment (gl2748) [reply src/app/components/cards/PostFull.scss]: No. Will restore.

File: src/app/components/elements/Voting.jsx [hunk 7]
```diff
@@ -68,53 +68,64 @@ class Voting extends React.Component {
...
+                permlink,
+                username,
+                is_comment,
+                updateWeight,
+                weight,
+                vote,
+                net_vesting_shares,
+                flag,
+            } = this.props;
+            if (net_vesting_shares > VOTE_WEIGHT_DROPDOWN_THRESHOLD) {
```
[the remark this answers is not in the document]
Comment (gl2748) [reply src/app/components/elements/Voting.jsx]: The inline comment notes that local storage is used here for preserving vote weight across tabs ... With the weight stored in redux, this is no longer necessary. Basically - Local storage for 'voteWeight' isn't being read anywhere.

I have made a commit to remove it.

File: src/app/components/elements/Voting.jsx [hunk 8]
```diff
@@ -127,7 +138,7 @@ class Voting extends React.Component {
             const { username, is_comment } = this.props;
             // Upon opening dialog, read last used weight (this works accross tabs)
             if (!this.state.showWeight) {
-                localStorage.removeItem('vote_weight'); // deprecated. remove this line after 8/31
+                localStorage.removeItem('vote_weight'); // deprecated. remove this      line after 8/31
```
[the remark this answers is not in the document]
Comment (gl2748) [reply src/app/components/elements/Voting.jsx]: Perhaps, `vote_weight` isn't even in localStorage though, it's `voteWeight` ... I think this is dead code.

File: src/app/components/elements/Voting.jsx (hunk 4 above)
Comment (gl2748) [reply src/app/components/elements/Voting.jsx]: I actually just moved this line up to be with the other React-Lifecycle methods. See Line 73 above...

File: src/app/components/pages/UserProfile.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (bnchdrff) [reply src/app/components/pages/UserProfile.jsx]: we should probably just use the value `g.author_rewards` for the value... this is a side issue though and not relevant to this work right now

File: src/app/components/elements/Voting.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (bnchdrff) [reply src/app/components/elements/Voting.jsx]: we should work to keep this feature if this is the case; i.e. we should set the value in localStorage in a redux saga.

File: src/app/components/elements/Voting.jsx (hunk 4 above)
Comment (bnchdrff) [reply src/app/components/elements/Voting.jsx]: ah, oops

File: src/app/components/pages/UserProfile.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (gl2748) [reply src/app/components/pages/UserProfile.jsx]: I agree. It's not very pure as it currently stands.

**Pull request merged**